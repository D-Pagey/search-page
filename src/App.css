.app {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.filter-div {
  background-color: #172B24;  
  bottom: 0;
  display: flex;
  justify-content: center;
  left: 0;
  margin-top: 16px;
  padding: 1rem;
  position: fixed;
  right: 0;
}

.filter-btn {
  width: 288px !important; 
  margin: 0 !important;
}