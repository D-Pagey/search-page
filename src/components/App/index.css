.app {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 0 auto 40px;
}

@media (min-width: 1024px) {
  .app {
    align-items: flex-start;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
}

.refine-search-div {
  align-items: center;
  background-color: #172B24;  
  bottom: 0;
  display: flex;
  justify-content: center;
  padding: 11px;
  position: fixed;
  width: 100%;
}

@media (min-width: 1024px) {
  .refine-search-div {
    display: none;
  }
}

.hide {
  display: none;
}

@media (min-width: 1024px) {
  .hide {
    display: block;
    margin-left: 32px;
    width: 25%;
  }
}