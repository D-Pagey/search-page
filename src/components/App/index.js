import React, { Component } from 'react';

import './index.css';
import Navbar from '../Navbar';
import Banner from '../Banner';
import Button from '../Button';
import Search from '../Search';
import Filter from '../Filter';
import Footer from '../Footer';

export default class App extends Component {
  state = {
    isRefineSearch: false,
    query: {
      location: '',
      start: '',
      distance: '',
      budget: '',
      gearbox: '',
      year: '',
      fuel: '',
    },
  }

  toggleRefineSearch = () => {
    const { isRefineSearch } = this.state;

    this.setState({ isRefineSearch: !isRefineSearch });
  }

  render() {
    const { isRefineSearch } = this.state;
    const buttonText = isRefineSearch ? 'Update search and hide filter' : 'Refine your search';

    return (
      <div className="app">
        <Navbar />
        <Banner />
        {isRefineSearch ? <Filter toggle={this.toggleRefineSearch} /> : <Search />}
        <Footer />
        <div className="refine-search-div">
          <Button onClick={this.toggleRefineSearch}>{buttonText}</Button>
        </div>
      </div>
    );
  }
}

// refine search here
// json object to query string
