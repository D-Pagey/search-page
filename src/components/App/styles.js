import styled from 'styled-components';

export const AppWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 0 auto 40px;

  @media (min-width: 1024px) {
    align-items: flex-start;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  } 
`;

export const RefineSearch = styled.div`
  align-items: center;
  background-color: #172B24;  
  bottom: 0;
  display: flex;
  justify-content: center;
  padding: 11px;
  position: fixed;
  width: 100%;

  @media (min-width: 1024px) {
    display: none;
  }
`;
