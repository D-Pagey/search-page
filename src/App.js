import React, { Component } from 'react';

import './App.css';
import Navbar from './components/presentational/navbar';
import Search from './components/presentational/search';

// const URL = 'https://app.joindrover.com/api/web/vehicles';

export default class App extends Component {
  state = {

  }

  render() {
    return (
      <div>
        <Navbar />
        <Search />
      </div>
    );
  }
}

/** To Do
 * Check all styling
 * Use feature branches
 * Icon
 * Menu bar
 * PropTypes
 * Default props
 */
