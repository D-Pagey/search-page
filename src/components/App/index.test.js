import React from 'react';
import { shallow } from 'enzyme';
import App from '.';

it('should render', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

// update ESLint for Jest .test files
