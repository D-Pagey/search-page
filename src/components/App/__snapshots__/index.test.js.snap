// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="app"
>
  <Navbar />
  <Banner />
  <Search />
  <Footer />
  <div
    className="refine-search-div"
  >
    <Button
      onClick={[Function]}
    >
      Refine your search
    </Button>
  </div>
</div>
`;
